src/ebusd/mqtthandler.cpp

```diff
@@ -860,21 +869,27 @@ bool MqttHandler::handleTraffic(bool allowReconnect) {
 
 string MqttHandler::getTopic(const Message* message, const string& suffix, const string& fieldName) {
   ostringstream ret;
-  for (size_t i = 0; i < g_topicStrs.size(); i++) {
-    ret << g_topicStrs[i];
-    if (!message) {
-      break;
-    }
-    if (i < g_topicFields.size()) {
-      if (g_topicFields[i] == "field") {
-        ret << fieldName;
-      } else {
-        message->dumpField(g_topicFields[i], false, &ret);
+  if (!g_azureIoTcompat) {
```

Comment:
this should be left as is

Reply:
I changed as requested, but it's not clear to me how to configure it on the command line. Azure IoT hub doesn't accept nested topics (only top level).